basics/form.php

```diff
@@ -0,0 +1,21 @@
+<?php
+require 'pdf.php';
+require 'print_data.php';
+require 'generate_pdf.php';
+if (isset($_POST["submit"])) {
+    $first_name = $_POST["first_name"];
+    $last_name = $_POST["last_name"];
```

Comment:
No need to declare these variables separately.
Directly use from $_POST.
Same for other places.
Directly use $_POST and do not assign to a variable unless you are using the variable again and again.